iroh-net: default features are not compatible with doc tests

```
$ cargo test -p iroh-net
```

fails with 
```
---- iroh-net/src/discovery.rs - discovery (line 73) stdout ----
error[E0432]: unresolved import `iroh_net::discovery::local_swarm_discovery`
   --> iroh-net/src/discovery.rs:76:26
    |
6   | use iroh_net::discovery::local_swarm_discovery::LocalSwarmDiscovery;
    |                          ^^^^^^^^^^^^^^^^^^^^^ could not find `local_swarm_discovery` in `discovery`
    |
note: found an item that was configured out
   --> /Users/dignified/opensource/iroh/iroh-net/src/discovery.rs:115:9
    |
115 | pub mod local_swarm_discovery;
    |         ^^^^^^^^^^^^^^^^^^^^^
note: the item is gated behind the `discovery-local-network` feature
   --> /Users/dignified/opensource/iroh/iroh-net/src/discovery.rs:113:7
    |
113 | #[cfg(feature = "discovery-local-network")]
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.

failures:
    iroh-net/src/discovery.rs - discovery (line 73)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -216,6 +216,7 @@ jobs:
     env:
       RUSTC_WRAPPER: "sccache"
       SCCACHE_GHA_ENABLED: "on"
+      RUSTDOCFLAGS: --cfg docsrs
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
@@ -226,8 +227,9 @@ jobs:
 
       - name: Docs
         run: cargo doc --workspace --all-features --no-deps --document-private-items
-        env:
-          RUSTDOCFLAGS: --cfg docsrs
+
+      - name: Docs (default features)
+        run: cargo doc --workspace --no-deps
 
   clippy_check:
     timeout-minutes: 30

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -96,6 +96,8 @@ jobs:
             targets="--lib --bins"
             echo cargo check -p $i $FEATURES $targets
             cargo check -p $i $FEATURES $targets
+            echo cargo test --doc --no-run $FEATURES
+            cargo test --doc --no-run $FEATURES
           done
         env:
           RUST_LOG: ${{ runner.debug && 'TRACE' || 'DEBUG'}}

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -96,8 +96,6 @@ jobs:
             targets="--lib --bins"
             echo cargo check -p $i $FEATURES $targets
             cargo check -p $i $FEATURES $targets
-            echo cargo test --doc --no-run $FEATURES
-            cargo test --doc --no-run $FEATURES
           done
         env:
           RUST_LOG: ${{ runner.debug && 'TRACE' || 'DEBUG'}}
@@ -129,13 +127,24 @@ jobs:
 
       - name: doctests
         if: ${{ (! inputs.flaky) &&  matrix.features == 'all' }}
+        env:
+          RUST_LOG: ${{ runner.debug && 'TRACE' || 'DEBUG'}}
         run: |
-          if [ -n "${{ runner.debug }}" ]; then
-              export RUST_LOG=TRACE
-          else
-              export RUST_LOG=DEBUG
-          fi
           cargo test --workspace --all-features --doc
+
+      # We can not check the doctest features by only compiling them,
+      # so are forced to run them once more.
+      - name: Check doctest features
+        if: ${{ ! inputs.flaky }}
+        env:
+          RUST_LOG: ${{ runner.debug && 'TRACE' || 'DEBUG'}}
+        run: |
+          for i in ${CRATES_LIST//,/ }
+          do
+            echo "Checking $i doctest $FEATURES"
+            echo cargo test -p $i --doc $FEATURES
+            cargo test -p $i --doc $FEATURES
+          done
 
   build_and_test_windows:
     timeout-minutes: 30

--- a/iroh-relay/src/server.rs
+++ b/iroh-relay/src/server.rs
@@ -138,8 +138,8 @@ pub struct QuicConfig {
     pub bind_addr: SocketAddr,
     /// The TLS server configuration for the QUIC server.
     ///
-    /// If this [`rustls::ServerConfig`] does not support
-    /// TLS 1.3, the QUIC server will fail at [`QuicServer::spawn`].
+    /// If this [`rustls::ServerConfig`] does not support TLS 1.3, the QUIC server will fail
+    /// to spawn.
     pub server_config: rustls::ServerConfig,
 }
 

--- a/iroh/src/discovery.rs
+++ b/iroh/src/discovery.rs
@@ -71,6 +71,8 @@
 //! [`ConcurrentDiscovery`]:
 //!
 //! ```no_run
+//! # #[cfg(feature = "discovery-local-network")]
+//! # {
 //! # use iroh::discovery::dns::DnsDiscovery;
 //! # use iroh::discovery::local_swarm_discovery::LocalSwarmDiscovery;
 //! # use iroh::discovery::pkarr::PkarrPublisher;
@@ -85,6 +87,7 @@
 //!     Box::new(LocalSwarmDiscovery::new(secret_key.public())?),
 //! ]);
 //! # Ok(())
+//! # }
 //! # }
 //! ```
 //!

--- a/iroh/src/discovery.rs
+++ b/iroh/src/discovery.rs
@@ -30,7 +30,6 @@
 //!
 //! - The [`PkarrResolver`] which can perform lookups from designated [pkarr relay servers]
 //!   using HTTP.
-//!
 #![cfg_attr(
     feature = "discovery-local-network",
     doc = "- [`LocalSwarmDiscovery`]: local_swarm_discovery::LocalSwarmDiscovery

--- a/iroh/src/discovery.rs
+++ b/iroh/src/discovery.rs
@@ -31,8 +31,11 @@
 //! - The [`PkarrResolver`] which can perform lookups from designated [pkarr relay servers]
 //!   using HTTP.
 //!
-//! - The [`LocalSwarmDiscovery`] discovers iroh nodes present on the local network,
-//!   very similar to mdNS.
+#![cfg_attr(
+    feature = "discovery-local-network",
+    doc = "- [`LocalSwarmDiscovery`]: local_swarm_discovery::LocalSwarmDiscovery
+             very similar to mDNS."
+)]
 //!
 //! - The [`DhtDiscovery`] also uses the [`pkarr`] system but can also publish and lookup
 //!   records to/from the Mainline DHT.
@@ -67,7 +70,13 @@
 //! # }
 //! ```
 //!
-//! To also enable [`LocalSwarmDiscovery`], it can be added as another service in the
+//! To also enable
+#![cfg_attr(feature = "discovery-local-network", doc = "[`LocalSwarmDiscovery`]")]
+#![cfg_attr(
+    not(feature = "discovery-local-network"),
+    doc = "`LocalSwarmDiscovery`"
+)]
+//! it can be added as another service in the
 //! [`ConcurrentDiscovery`]:
 //!
 //! ```no_run
@@ -97,9 +106,12 @@
 //! [Number 0]: https://n0.computer
 //! [`PkarrResolver`]: pkarr::PkarrResolver
 //! [`PkarrPublisher`]: pkarr::PkarrPublisher
-//! [`LocalSwarmDiscovery`]: local_swarm_discovery::LocalSwarmDiscovery
 //! [`DhtDiscovery`]: pkarr::dht::DhtDiscovery
 //! [pkarr relay servers]: https://pkarr.org/#servers
+#![cfg_attr(
+    feature = "discovery-local-network",
+    doc = "[`LocalSwarmDiscovery`]: local_swarm_discovery::LocalSwarmDiscovery"
+)]
 
 use std::time::Duration;
 

